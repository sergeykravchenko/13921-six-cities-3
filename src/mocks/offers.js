export default [
  {
    "id": 1,
    "zoom": 15,
    "name": `Canal View Prinsengracht`,
    "picture": `img/room.jpg`,
    "price": 120,
    "priceText": `night`,
    "rating": 4.8,
    "features": {
      "type": `Private room`,
      "bedrooms": 3,
      "maxGuests": 3,
    },
    "isPremium": true,
    "isInBookmark": true,
    "houseHolds": [
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
      `Heating`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`,
      `Dishwasher`,
      `Cabel TV`,
      `Fridge`,
    ],
    "images": [
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
      `img/room.jpg`,
      `img/studio-01.jpg`,
      `img/apartment-03.jpg`,
    ],
    "host": {
      "id": 2,
      "name": `Max`,
      "img": `img/avatar-max.jpg`,
      "isPro": true,
    },
    "description": `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
    "coords": [52.3909553943508, 4.85309666406198],
    "reviews": [
      {
        "id": 1,
        "author": `Max`,
        "avatar": `img/avatar-max.jpg`,
        "rating": 4,
        "text": `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
                 The building is green and from 18th century.`,
        "date": new Date(`2019-04-10`),
      }, {
        "id": 2,
        "author": `Angelina`,
        "avatar": `img/avatar-angelina.jpg`,
        "rating": 5,
        "text": `Picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
        "date": new Date(`2019-04-10`),
      }
    ],
    "city": {
      "name": `Amsterdam`,
      "coords": [52.38333, 4.9],
    },
  }, {
    "id": 2,
    "zoom": 13,
    "name": `Nice, cozy, warm big bed apartment`,
    "picture": `img/apartment-02.jpg`,
    "price": 80,
    "priceText": `night`,
    "rating": 5,
    "features": {
      "type": `Apartment`,
      "bedrooms": 3,
      "maxGuests": 3,
    },
    "houseHolds": [
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
      `Heating`,
      `Coffee machine`,
    ],
    "images": [
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
      `img/room.jpg`,
      `img/studio-01.jpg`,
      `img/apartment-03.jpg`,
    ],
    "host": {
      "id": 3,
      "name": `Angelina`,
      "img": `img/avatar-angelina.jpg`,
      "isPro": false,
    },
    "description": `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    "coords": [52.369553943508, 4.85309666406198],
    "reviews": [
      {
        "id": 1,
        "author": `Angelina`,
        "avatar": `img/avatar-angelina.jpg`,
        "rating": 4.5,
        "text": `Picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
        "date": new Date(`2019-04-8`),
      }
    ],
    "city": {
      "name": `Amsterdam`,
      "coords": [52.38333, 4.9],
    },
  }, {
    "id": 3,
    "zoom": 11,
    "name": `Wood and stone place`,
    "picture": `img/room.jpg`,
    "price": 132,
    "priceText": `night`,
    "rating": 5,
    "features": {
      "type": `Private room`,
      "bedrooms": 4,
      "maxGuests": 3,
    },
    "isPremium": true,
    "houseHolds": [
      `Baby seat`,
      `Kitchen`,
      `Dishwasher`,
      `Cabel TV`,
    ],
    "images": [
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
      `img/room.jpg`,
      `img/studio-01.jpg`,
      `img/apartment-03.jpg`,
    ],
    "host": {
      "id": 4,
      "name": `Max`,
      "img": `img/avatar-max.jpg`,
      "isPro": false,
    },
    "description": `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
    "coords": [52.3909553943508, 4.929309666406198],
    "reviews": [
      {
        "id": 1,
        "author": `Max`,
        "avatar": `img/avatar-max.jpg`,
        "rating": 4,
        "text": `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
                 The building is green and from 18th century.`,
        "date": new Date(`2019-04-24`),
      }
    ],
    "city": {
      "name": `Cologne`,
      "coords": [52.38343, 4.9],
    }
  }, {
    "id": 4,
    "zoom": 18,
    "name": `Beautiful &amp; luxurious apartment at great location`,
    "picture": `img/apartment-01.jpg`,
    "price": 180,
    "priceText": `night`,
    "rating": 4,
    "features": {
      "type": `Apartment`,
      "bedrooms": 5,
      "maxGuests": 3,
    },
    "houseHolds": [
      `Heating`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`,
    ],
    "images": [
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
      `img/room.jpg`,
      `img/studio-01.jpg`,
      `img/apartment-03.jpg`,
    ],
    "host": {
      "id": 1,
      "name": `Angelina`,
      "img": `img/avatar-angelina.jpg`,
      "isPro": true,
    },
    "description": `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    "coords": [52.3809553943508, 4.939309666406198],
    "city": {
      "name": `Hamburg`,
      "coords": [52.38013, 4.9],
    },
  },
];
