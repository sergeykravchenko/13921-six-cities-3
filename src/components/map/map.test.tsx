import * as React from "react";
import * as renderer from "react-test-renderer";
import Map from "./map";
import offers from "../../mocks/offers";

const string = `className`;

const activeCity = {
  "name": `Amsterdam`,
  "coords": [52.38333, 4.9],
  "zoom": 14,
};

const activeMarker = 2;
const zoom = 15;

it(`Should map render correctly`, () => {
  const tree = renderer
    .create(<Map
      bemBlock={string}
      coords={activeCity.coords}
      activeMarker={activeMarker}
      offers={offers}
      zoom={zoom}
    />, {
      createNodeMock: () => {
        return document.createElement(`div`);
      }})
    .toJSON();

  expect(tree).toMatchSnapshot();
});
