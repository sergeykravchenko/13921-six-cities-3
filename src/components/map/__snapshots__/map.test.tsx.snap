// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should map render correctly 1`] = `
<section
  className="className__map map"
/>
`;
